scapy/arch/bpf/supersocket.py

```diff
@@ -390,14 +390,23 @@ def send(self, pkt):
             self.assigned_interface = iff
 
         # Build the frame
-        if self.guessed_cls == Loopback:
-            # bpf(4) man page (from macOS, but also for BSD):
+        if self.guessed_cls == Loopback and not (DARWIN and
```

Comment:
This should probably be moved outside thé `send` function

Reply:
Yup, this is from #3257 so I'll need to change it there.